modules/utils/src/merkle.ts

```diff
@@ -49,7 +49,7 @@ export const addTransferToTree = (transfer: CoreTransferState, tree: merkle.Tree
 export const removeTransferFromTree = (transfer: CoreTransferState, tree: merkle.Tree): MerkleTreeUpdate => {
   let root: string;
   try {
-    tree.insert_hex_js(encodeCoreTransferState(transfer));
+    tree.delete_id_js(transfer.transferId);
```

Comment:
This and other places are still calling `tree.free` at the wrong time.

Remember: You cannot call any method on `tree` after calling `free`.

The right time to call `free` would be after calling `this.trees.delete(channelAddress)`. Since this doesn't seem to happen, just don't call free at all.